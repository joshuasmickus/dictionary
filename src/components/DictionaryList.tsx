import * as React from 'react';
import { connect } from 'react-redux';

import { IDictionaries, IDictionary } from 'src/state/dictionary/dictionaryReducer';

interface IDictionaryListProps {
  list: IDictionary[];
};

interface IDictionaryState {
  dictionary: IDictionaries;
}

const DictionaryList: React.FC<IDictionaryListProps> = ({ list }) => {
  const deleteItem = (item: IDictionary['id']) => () => {
    console.log('delete: ', item); // tslint:disable-line
  };
  const viewItem = (item: IDictionary['id']) => () => {
    console.log('view: ', item); // tslint:disable-line
  };

  return (
    <ul>
      {
        list.map((dictionary: IDictionary) =>
          (
            <li key={dictionary.id}>
              <a onClick={viewItem(dictionary.id)}>{dictionary.from}/{dictionary.to}</a> -
              <a onClick={deleteItem(dictionary.id)}>delete</a>
            </li>
          )
        )
      }
    </ul>
  )
};

const mapStateToProps = ({ dictionary }: IDictionaryState) => ({
  list: dictionary.list
});

export default connect(mapStateToProps)(DictionaryList);
