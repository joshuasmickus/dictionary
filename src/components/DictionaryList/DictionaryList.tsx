import * as React from 'react';
import { Link } from 'react-router-dom';

import { IDictionaryName, IDictionaryState } from 'src/models/common';
import { deleteDictionary } from 'src/state/dictionary/dictionaryRoutines';

interface IDictionaryListProps {
  list: IDictionaryName[];
  deleteDictionaryRequest: (id: string) => void;
};

export const DictionaryList: React.FC<IDictionaryListProps> = ({
  list,
  deleteDictionaryRequest
}) => {
  const deleteItem = (id: IDictionaryName['id']) => (event: React.MouseEvent) => {
    event.preventDefault();

    deleteDictionaryRequest(id);
  };

  return (
    <ul>
      {
        list.length > 0 && list.map((dictionary: IDictionaryName) =>
          (
            <li key={dictionary.id}>
              {dictionary.name}&nbsp;
              <Link to={`/edit-dictionary/${dictionary.id}`}>edit</Link> -&nbsp;
              <a onClick={deleteItem(dictionary.id)} href={`/delete-dictionary/${dictionary.id}`}>delete</a>
            </li>
          )
        )
      }
      { list.length === 0 && <li>There are no dictionaries.</li> }
    </ul>
  )
};

export const mapStateToProps = ({ dictionary }: IDictionaryState) => ({
  list: dictionary.list
});

export const mapDispatchToProps = (dispatch: (action: any) => void) => ({
  deleteDictionaryRequest: (id: IDictionaryName['id']) => dispatch(deleteDictionary.request(id))
});
