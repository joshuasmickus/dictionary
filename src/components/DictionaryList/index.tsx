import { connect } from 'react-redux';

import {
  DictionaryList,
  mapDispatchToProps,
  mapStateToProps
} from './DictionaryList';

export default connect(mapStateToProps, mapDispatchToProps)(DictionaryList);
