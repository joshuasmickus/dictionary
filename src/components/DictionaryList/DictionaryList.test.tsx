import { createMemoryHistory } from 'history'
import * as React from 'react';
import { Router } from 'react-router-dom';
import { render } from 'react-testing-library';

import { DictionaryList } from './DictionaryList';

const id = 'someid';
const requiredProps = {
  deleteDictionaryRequest: jest.fn(),
  list: [{ name: 'something', id }],
};

describe('Given a DictionaryList component', () => {
  describe('when it is rendered', () => {
    it('should match the snapshot', () => {
      const { container } = render(<Router history={createMemoryHistory()}><DictionaryList {...requiredProps} /></Router>);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
