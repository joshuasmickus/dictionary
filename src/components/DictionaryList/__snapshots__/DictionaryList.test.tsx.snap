// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a DictionaryList component when it is rendered should match the snapshot 1`] = `
<ul>
  <li>
    something
    <a
      href="/edit-dictionary/someid"
    >
      edit
    </a>
    -Â 
    <a
      href="/delete-dictionary/someid"
    >
      delete
    </a>
  </li>
</ul>
`;
